debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/stackwalker/StackWalker.java

```diff
@@ -836,28 +870,27 @@ public FrameCallbackResult walkFrame(WalkState walkState)
 						break walkIt;
 					}
 				}
-	
+
 				if ((walkState.flags & J9_STACKWALK_ITERATE_FRAMES) != 0) {
 					break walkIt;
 				}
-	
+
 				return FrameCallbackResult.KEEP_ITERATING;
 			} while (false);
-	
+
 			// walkIt:
 			if ((walkState.flags & J9_STACKWALK_ITERATE_FRAMES) != 0) {
-				FrameCallbackResult rc = walkState.callBacks.frameWalkFunction(
-						walkState.walkThread, walkState);
-	
+				FrameCallbackResult rc = walkState.callBacks.frameWalkFunction(walkState.walkThread, walkState);
```

Comment:
While you did fix the use of `walkState.walkThread` in the actual frameWalkFunction, I feel this could lead to future problems if we don't specify how walkThread can be null here. To me this parameter is redundant and should be removed since callback can still get the walkThread pointer via `walkState.walkThread` where it should do the proper null check.

same apply for `objectSlotWalkFunction`